ts/Extensions/Annotations/Popup.ts

```diff
@@ -830,127 +876,321 @@ H.Popup.prototype = {
+         * Filter object of series which are not indicators.
+         * If the filter string exists, check against it.
+         *
+         * @private
+         *
+         * @param {Highcharts.FilteredSeries} series
+         *        All series are available in the plotOptions.
+         *
+         * @param {string|undefined} filter
+         *        Applied filter string from the input.
+         *        For the first iteration, it's an empty string.
+         *
+         * @return {void}
+         */
+        filterSeries: function (
+            this: Highcharts.Popup,
+            series: Highcharts.FilteredSeries,
+            filter?: string
+        ): [[string, string, Series]] {
+            const popup = this,
+                indicators = popup.indicators;
+            let filteredSeries: [[string, string, Series]] = [] as any;
+
+            objectEach(series, function (
+                series: Series,
+                value: string
+            ): void {
+                const seriesOptions = series.options;
+                // Allow only indicators.
+                if ((series as any).params || seriesOptions && (seriesOptions as any).params) {
+                    const { indicatorFullName, indicatorType } = indicators.getNameType(series, value);
+
+                    if (filter) {
+                        const filterLength = filter.length,
+                            slicedValue = indicatorFullName.slice(0, filterLength);
+                        if (slicedValue.toLocaleLowerCase() === filter.toLocaleLowerCase()) {
+                            filteredSeries.push([indicatorFullName, indicatorType, series]);
+                        }
+                    } else {
+                        filteredSeries.push([indicatorFullName, indicatorType, series]);
+                    }
+                }
+            });
+
+            // Sort indicators alphabeticaly.
+            return filteredSeries.sort(function (a, b): number {
+                const seriesAName = a[0].toLowerCase(),
+                    seriesBName = b[0].toLowerCase();
+                return (seriesAName < seriesBName) ? -1 : (seriesAName > seriesBName) ? 1 : 0;
+            });
+        },
+
+        /**
+         * Filter an array of series and map its names and types.
+         *
+         * @private
+         *
+         * @param {Highcharts.FilteredSeries} series
+         *        All series are available in the plotOptions.
+         *
+         * @return {void}
+         */
+        filterSeriesArray: function (
+            this: Highcharts.Popup,
+            series: Array<Series>
+        ): [[string, string, Series]] {
+            let filteredSeries: [[string, string, Series]] = [] as any;
+
+            // Allow only indicators.
+            series.forEach(function (series): void {
+                const seriesOptions = series.options;
+
+                if ((series as any).params || seriesOptions && (seriesOptions as any).params) {
```

Comment:
`series.is('sma')`

Reply:
Here as the series is rendered I can check its' type.